### DISCUSSION TITLE
HTTP3/HTTP2 usage in Client

### DISCUSSION THREAD

**Discussion by SbiCA** [ravendb/ravendb#18655] [General]
Hey,

I was wondering if the current client lib is setting up the pipeline using http3 or http2? 
https://learn.microsoft.com/en-us/dotnet/core/extensions/httpclient-http3 

Since that should give quite some perf boost but I was wondering if the benchmarks should show this?

**Comment by ayende** [comment]
We'll use `HTTP2` automatically on Posix platforms or on Windows 10 or higher.

**Reply by SbiCA** [reply]
👌  and did you already do an experiment with http3? I'd be interested to understand if there is an overall perf gain.

As I can see there is still an open issue https://issues.hibernatingrhinos.com/issue/RavenDB-17220/Support-for-HTTP-3-in-Kestrel

**Reply by ppekrol** [reply]
Not yet, something that we will definitely do later this year, but from prev experience it is always good to wait at least one .NET version update before adoption.

**Reply by SbiCA** [reply]
Nice good to hear we ll get some insights soon then since it's almost November 😄
